go/pkg/tool/tool.go

```diff
@@ -326,18 +326,45 @@ func (c *Client) DownloadDirectory(ctx context.Context, rootDigest, path string)
 	return err
 }
 
+// UploadStats contains various metadata of a directory upload.
+type UploadStats struct {
+	rc.TreeStats
+	RootDigest       digest.Digest
+	TotalNumBlobs    int64
+	NumCacheMisses   int64
+	BytesTransferred int64
+	CacheMissesBytes int64
```

Comment:
Nit: I find shared prefixes easier to read and helps auto-complete engines with suggestions.

```suggestion
	CountBlobs    int64
	CountCacheMisses  int64
	BytesTransferred int64
	BytesCacheMisses int64
```

Reply:
Done.